OpenMTR/Utilities/ImageUtils.cs

```diff
@@ -11,27 +11,27 @@ public static class ImageUtils
     {
         public static void ColorToGray(DataObject dataObject)
         {
-            Cv2.CvtColor(dataObject.Image, dataObject.Image, ColorConversionCodes.BGR2GRAY);
+            Cv2.CvtColor(dataObject.SourceImage, dataObject.ModifiedImage, ColorConversionCodes.BGR2GRAY);
         }
 
         public static void ColorToGray(List<DataObject> dataObjectList)
         {
             foreach (DataObject dataObject in dataObjectList)
             {
-                Cv2.CvtColor(dataObject.Image, dataObject.Image, ColorConversionCodes.BGR2GRAY);
+                Cv2.CvtColor(dataObject.SourceImage, dataObject.ModifiedImage, ColorConversionCodes.BGR2GRAY);
             }
         }
 
         public static void ApplyGaussianBlur(DataObject dataObject)
         {
-            Cv2.GaussianBlur(dataObject.Image, dataObject.Image, new Size(3, 3), 0, 0, BorderTypes.Default);
+            Cv2.GaussianBlur(dataObject.SourceImage, dataObject.ModifiedImage, new Size(3, 3), 0, 0, BorderTypes.Default);
```

Comment:
How about instead of taking in `DataObject`, we take in two `Mat` objects? Then we have the freedom of doing in whatever order. 
`public static void ApplyGaussianBlur(Mat sourceImage, Mat destinationImage)`
`public static void ColorToGray(Mat sourceImage, Mat destinationImage)`
```C#
ColorToGray(dataObject.SourceImage, dataObject.ModifiedImage);
ApplyGaussianBlur(dataObject.ModifiedImage, dataObject.ModifiedImage);
```

Reply:
that will work!

Reply:
You know what, let me take a quick look using a raster editor and I'll see what sort of difference it makes to apply a gaussian blur before and after going greyscale

Reply:
@cclinard84 I'll do that.

Reply:
I don't think there is too much of a change either way @xervir. It's generally just used to soften blurry lines to make them more pronounced for a canny filter to pick up on.